### DISCUSSION TITLE
Struggling with 'dotnet pack" and loading 'directory.build.props'

### DISCUSSION THREAD

**Discussion by masterzdran** [community/community#26272] [Actions]
<p>Greetings,</p>
<p>I hope that you are feeling well.</p>
<p>Well, I’m struggling within “Github Actions”, :stuck_out_tongue: .</p>
<p>I have a dotnet library solution, and I expect to end up with a versioned nuget package to push to <a href="http://nuget.org" rel="noopener nofollow ugc">nuget.org</a>. Simple right?! :stuck_out_tongue:</p>
<p>Actually, not really.</p>
<p>I’m going to focus on the “dotnet pack” challenge. There are two challenges :slight_smile:</p>
<p>I use <strong>directory.build.props</strong> to define a set of properties that “dotnet pack” will use to generate a versioned nuget package, with the most common fields filled. If a do a</p>
<pre><code class="lang-auto">dotnet pack --configuration Release
</code></pre>
<p>on my windows and ubuntu command line, it works as expected.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/8/98fa94c61ecf265cc05385fdec46917923fe9bc6.png" data-download-href="/uploads/short-url/lPjrtr98ndjHS9UodeZKhPCvd9s.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2105110/181096578-cd6c235f-60d0-4838-b2c0-90ac2e2148c4.png" alt="image" data-base62-sha1="lPjrtr98ndjHS9UodeZKhPCvd9s" width="690" height="21" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/8/98fa94c61ecf265cc05385fdec46917923fe9bc6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1887×58 5.38 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>However on <a href="http://Giihub.com" rel="noopener nofollow ugc">Giihub.com</a><br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/5/456d4f6ae0d07bc7df557f06989d4bab4aa5e0fe.png" data-download-href="/uploads/short-url/9Ub8M4f0vnHxGTRYdYJ964GTI3I.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2105110/181096581-2d57f9aa-94a6-4026-8191-6a03ec629ebf.png" alt="image" data-base62-sha1="9Ub8M4f0vnHxGTRYdYJ964GTI3I" width="690" height="51" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/5/456d4f6ae0d07bc7df557f06989d4bab4aa5e0fe_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1294×97 8.43 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>The package is generated but without the version suffix.</p>
<p>This is my first issue.<br>
As a “workaround,” I added a parameter to the step action:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/0/3034076d23d22aadf0b530ce405a9f988eb46565.png" data-download-href="/uploads/short-url/6Sqoc0dtW5Dug7sQJcTskV7xjs9.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2105110/181096582-cf63fff7-bd4c-4866-bccc-8ff137fac5fe.png" alt="image" data-base62-sha1="6Sqoc0dtW5Dug7sQJcTskV7xjs9" width="690" height="82" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/0/3034076d23d22aadf0b530ce405a9f988eb46565_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">940×113 4.02 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>It works, but I rather don’t have it. That was the purpose of the <strong>directory.build.props</strong>.</p>
<p>My second issue, the package generated doesn’t have the properties defined on <strong>directory.build.props</strong>.<br>
This is the package generated on <a href="http://Github.com" rel="noopener nofollow ugc">Github.com</a>.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/f/7/f76ca4b90332fca5249f47daa7532298b6176cc3.png" data-download-href="/uploads/short-url/ziOHzeDhaIAq8ajfSR5tAbVV7YT.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2105110/181096583-d31337c0-0a46-467a-90a3-054b5e41a780.png" alt="image" data-base62-sha1="ziOHzeDhaIAq8ajfSR5tAbVV7YT" width="690" height="226" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/f/7/f76ca4b90332fca5249f47daa7532298b6176cc3_2_690x226.png, https://aws1.discourse-cdn.com/github/optimized/3X/f/7/f76ca4b90332fca5249f47daa7532298b6176cc3_2_1035x339.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/f/7/f76ca4b90332fca5249f47daa7532298b6176cc3_2_1380x452.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/f/7/f76ca4b90332fca5249f47daa7532298b6176cc3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1387×455 19.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>When the expectation was this:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/6/7682f6e6cf68f12723da098ecd9ae0e8e6d9dfc5.png" data-download-href="/uploads/short-url/gUoVKXXTbvC2gNOQfQR2aCZgcap.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2105110/181096584-635deb27-9add-489a-a974-28c111e446d4.png" alt="image" data-base62-sha1="gUoVKXXTbvC2gNOQfQR2aCZgcap" width="538" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/6/7682f6e6cf68f12723da098ecd9ae0e8e6d9dfc5_2_538x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/7/6/7682f6e6cf68f12723da098ecd9ae0e8e6d9dfc5_2_807x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/7/6/7682f6e6cf68f12723da098ecd9ae0e8e6d9dfc5.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/6/7682f6e6cf68f12723da098ecd9ae0e8e6d9dfc5_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">946×878 35.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>As a “workaround,” I moved all parameters to csproj. It works, but I rather don’t have it. That was the purpose of the <strong>directory.build.props</strong>.</p>
<p>Does anyone have a similar issue? Am I doing something wrong with my Github actions?</p>
<p>BTW, and it is my fault, I should confirm the package first:<br>
Now I have a “production package” on <strong><a href="http://nuget.org" rel="noopener nofollow ugc">nuget.org</a></strong> that should be a “pre-release package,” and the nuget box itself is very lame :frowning:</p>
<p>I appreciate all the constructive comments</p>

**Comment by masterzdran** [comment, marked as answer]
<p>Hi.</p>
<p>I figure out the issue. :slight_smile:</p>
<p>The problem is related to the name file, which is all lower case. The file is named " <strong>directory.build.props</strong>" when it should be named “** Directory.Build.props**”.</p>
<p>For future reference:<br>
<a href="https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build?view=vs-2022#directorybuildprops-and-directorybuildtargets" rel="noopener nofollow ugc">Customize your build - MSBuild | Microsoft Docs</a></p>
<p>It post may be closed. :stuck_out_tongue:</p>

**Reply by Xeevis** [reply]
You've got to be kidding me, did I just waste several hours because my file is named `Directory.build.props` instead of `Directory.Build.props`?  🤯🤕 Many thanks for sharing this bit of information, saving me tons of further frustration.

**Reply by PetesBreenCoding** [reply]
Hello, just to let you know I, 4 years on from this, have just encountered the same issue. Was banging my head (and Copilot's head) against it for ages. Worked locally just not in GitHub. All because I had a small "b" in "build". Unbelievable!
